test/test_mpi_p4est_3d.jl

```diff
@@ -0,0 +1,60 @@
+module TestExamplesMPIP4estMesh3D
+
+using Test
+using Trixi
+
+include("test_trixi.jl")
+
+# pathof(Trixi) returns /path/to/Trixi/src/Trixi.jl, dirname gives the parent directory
+const EXAMPLES_DIR = joinpath(pathof(Trixi) |> dirname |> dirname, "examples", "p4est_3d_dgsem")
+
+@testset "P4estMesh MPI 3D" begin
+
+# Run basic tests
+@testset "Examples 3D" begin
+  # Linear scalar advection
+  @trixi_testset "elixir_advection_basic.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_basic.jl"),
+      # Expected errors are exactly the same as with TreeMesh!
+      l2   = [0.00016263963870641478],
+      linf = [0.0014537194925779984])
+  end
+
+  @trixi_testset "elixir_advection_restart.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_restart.jl"),
+      l2   = [0.002590388934758452],
+      linf = [0.01840757696885409])
+  end
+
+  @trixi_testset "elixir_advection_cubed_sphere.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_cubed_sphere.jl"),
+      l2   = [0.002006918015656413],
+      linf = [0.027655117058380085])
+  end
+
+  # Compressible Euler
+  @trixi_testset "elixir_euler_source_terms_nonperiodic.jl" begin
```

Comment:
Are these numbers also identical to TreeMesh results?

Reply:
And more generally, are these results truly identical to the serial runs?

Reply:
We only did that for the `TreeMesh`. The `P4estMesh` can't reuse the same error calculation functions. In this case it's also not possible to make them truly identical because the serial code calculates the fluxes at the interfaces using the normal vectors obtained from the primary side but in the parallel case the locally available side is used (which might be the secondary side) and as a result the fluxes are only equal up to rounding errors.

Reply:
Expanding on my previous comment, another consequence of this is that if two processes share an MPI interface, the flux they both calculate for this MPI interface is only equal up to rounding errors as well. In particular this means, that not just the L2 but also the Linf errors are not identical (in a binary sense).

Besides the obvious performance penalty of calculating fluxes twice, this might be another argument in favor of a different approach to MPI interface flux calculation.

Reply:
Does that mean that the normal vectors are not consistent at a single interface point when looking from two different sides? Isn't that a problem in itself?

Reply:
Yes, that's correct. I think this is inherent to curved meshes due to the way the Jacobians and so on are calculated. I don't see how you could reasonably change that to ensure binary equality.